utils.js

```diff
@@ -1,10 +1,52 @@
 const settings = require("config.settings")
 
 const u = {
-    filterCS: (room, structureType) => {
-        const structures = room.find(FIND_STRUCTURES)
-        return structures.filter(structure => structure.structureType === structureType)
-    },
+    construction: {
+        filterCS: (room, structureType) => {
+            const structures = room.find(FIND_STRUCTURES)
+            return structures.filter(structure => structure.structureType === structureType)
+        },
+        findValidPositionNearby: (room, x, y) => {
+            const closeBy = false
+            /* 
+            XYPermutations: 
+            (1)  (2)  (3)
+            (4)  (x)  (5)
+            (6)  (7)  (8)
+            */
+            const XYPermutations = [
+                [x - 1, y],
+                [x - 1, y - 1],
+                [x - 1, y + 1],
+                [x, y - 1],
+                [x, y + 1],
+                [x + 1, y],
+                [x + 1, y - 1],
+                [x + 1, y + 1]
+            ]
+            for (const position in XYPermutations) {
+                let [_x, _y] = position
+                const observedArea = room.lookAt(_x, _y)
+                if (observedArea[0].type === 'terrain' && observedArea[0].terrain === 'plain') {
+                    return [_y, _x]
```

Comment:
```suggestion
                    return [_x, _y] 
```
Is that sequence reversal intended? Not sure of your intent there.

Reply:
lol I'm such a derp - you're right. !hanks!